src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -105,38 +90,78 @@ internal void ValidateCreateContext(bool isServer, string targetHost, SslProtoco
                 throw new InvalidOperationException(SR.net_auth_reauth);
             }
 
-            if (Context != null && IsServer != isServer)
+            if (Context != null && IsServer)
             {
                 throw new InvalidOperationException(SR.net_auth_client_server);
             }
 
-            if (targetHost == null)
+            if (sslClientAuthenticationOptions.TargetHost == null)
+            {
+                throw new ArgumentNullException(nameof(sslClientAuthenticationOptions.TargetHost));
+            }
+
+            if (sslClientAuthenticationOptions.TargetHost.Length == 0)
+            {
+                sslClientAuthenticationOptions.TargetHost = "?" + Interlocked.Increment(ref s_uniqueNameInteger).ToString(NumberFormatInfo.InvariantInfo);
```

Comment:
I'm not sure. @davidsh ?

Reply:
>I'm not sure. @davidsh ?

I don't know. The code is very old.  It's trying to prevent an empty string from being used as TargetHost, so it makes one up.